--- cipe-1.5.2/pkcipe/Makefile.in	Sat Dec 16 19:27:35 2000
+++ cipe-1.5.2-p/pkcipe/Makefile.in	Wed Nov  7 16:19:10 2001
@@ -40,12 +40,12 @@
 	-mkdir -p $(bindir) $(sbindir)
 	$(INSTALL) -m 755 rsa-keygen $(bindir)
 	$(INSTALL) -m 755 pkcipe $(sbindir)
-	[ -d /etc/cipe/pk ] || mkdir -p /etc/cipe/pk
-	[ -d /var/run/cipe ] || mkdir -p /var/run/cipe
-	chmod 700 /etc/cipe /etc/cipe/pk /var/run/cipe
-	@if [ ! -f /etc/cipe/identity.priv ] ; then \
+	[ -d $(SYSCONFDIR)/pk ] || mkdir -p $(SYSCONFDIR)/pk
+	[ -d $(LOCALSTATEDIR)/run/cipe ] || mkdir -p $(LOCALSTATEDIR)/run/cipe
+	chmod 700 $(SYSCONFDIR) $(SYSCONFDIR)/pk $(LOCALSTATEDIR)/run/cipe
+	@if [ ! -f $(SYSCONFDIR)/identity.priv ] ; then \
 		echo "*** Generating new identity (host) key ***" ;\
-		rsa-keygen /etc/cipe/identity ;\
+		echo rsa-keygen $(SYSCONFDIR)/identity ;\
 	fi
 
 clean:
diff -Naur cipe-1.5.2/conf/Makefile-obj.in cipe-1.5.2-p/conf/Makefile-obj.in
--- cipe-1.5.2/conf/Makefile-obj.in	Fri Apr 26 10:29:53 2002
+++ cipe-1.5.2-p/conf/Makefile-obj.in	Fri Apr 26 10:22:33 2002
@@ -40,6 +40,8 @@
 
 all::	$(MODULE) $(CIPED)
 
+modules:: $(MODULE)
+
 install:: all
 	-mkdir -p $(MODDIR) $(BINDIR)
 	install -m 644 $(MODULE) $(MODDIR)
diff -Naur cipe-1.5.2/conf/Makefile-top.in cipe-1.5.2-p/conf/Makefile-top.in
--- cipe-1.5.2/conf/Makefile-top.in	Fri Apr 26 10:30:00 2002
+++ cipe-1.5.2-p/conf/Makefile-top.in	Fri Apr 26 10:22:22 2002
@@ -23,6 +23,9 @@
 
 default: all
 
+modules:
+	$(MAKE) -C $(BUILD) modules
+
 SRCS:=	cipe.h crypto.h \
 	ciped.c ciped.h ioctl.c ioctl.h options.in genoptions.pl \
 	module.c device.c sock.c output.c encaps.c \
